lib/src/ui/puzzle/puzzle_history_screen.dart

```diff
@@ -0,0 +1,295 @@
+
+class PuzzleHistoryWidget extends ConsumerWidget {
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final historyState = ref.watch(puzzleHistoryProvider);
+    return historyState.when(
+      data: (data) {
+        if (data.isLoading) {
+          return Shimmer(
+            child: ShimmerLoading(
+              isLoading: true,
+              child: ListSection.loading(
+                itemsNumber: 5,
+                header: true,
+              ),
+            ),
+          );
+        }
+        if (!data.isLoading && data.historyList.isEmpty) {
+          return ListSection(
+            header: Text(context.l10n.puzzleHistory),
+            children: [
+              Center(
+                child: Text(context.l10n.puzzleNoPuzzlesToShow),
+              )
+            ],
+          );
+        }
+        final crossAxisCount =
+            MediaQuery.of(context).size.width > kTabletThreshold ? 4 : 2;
+        final boardWidth = (defaultTargetPlatform == TargetPlatform.iOS)
+            ? (MediaQuery.of(context).size.width) * 0.91 / crossAxisCount
+            : MediaQuery.of(context).size.width / crossAxisCount;
+        return ListSection(
+          header: Text(context.l10n.puzzleHistory),
+          headerTrailing: (data.historyList.length == 10)
+              ? NoPaddingTextButton(
+                  onPressed: () => pushPlatformRoute(
+                    context,
+                    builder: (context) => PuzzleHistoryScreen(data.historyList),
+                  ),
+                  child: Text(
+                    context.l10n.more,
+                  ),
+                )
+              : null,
+          children: [
+            for (var i = 0; i < data.historyList.length; i += crossAxisCount)
+              Row(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: data.historyList
+                    .getRange(
+                      i,
+                      i + crossAxisCount <= data.historyList.length
+                          ? i + crossAxisCount
+                          : data.historyList.length,
+                    )
+                    .map(
+                      (e) => _HistoryBoard(e, boardWidth),
+                    )
+                    .toList(),
+              ),
+          ],
+        );
+      },
+      error: (e, s) {
+        debugPrint(
+          'SEVERE: [PuzzleHistoryWidget] could not load dashboard',
+        );
+        return const Center(child: Text('Could not load Puzzle History'));
+      },
+      loading: () => Shimmer(
+        child: ShimmerLoading(
+          isLoading: true,
+          child: ListSection.loading(
+            itemsNumber: 5,
+            header: true,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class PuzzleHistoryScreen extends StatelessWidget {
+  const PuzzleHistoryScreen(this.historyList);
+
+  final IList<HistoryPuzzle> historyList;
```

Comment:
I don't think we should have this parameter. If the reason is to avoid fetching twice the data, we should use riverpod caching mechanism instead (if needed).